src/components/authpage/SignUpPage.jsx

```diff
@@ -0,0 +1,131 @@
+import React, { useState } from "react";
+import { NavLink } from "react-router-dom";
+import { useAuth } from "../../context/AuthProvider";
+import { useDocumentTitle } from "../../utils/usDocumentTitle";
+
+export default function SignUpPage() {
+    const [email, setEmail] = useState("");
+    const [emailError, setEmailError] = useState(false);
+    const [passwordError, setPasswordError] = useState(false);
+    const [password, setPassword] = useState("");
+    const [acceptTerms, setAcceptTerms] = useState(false);
+    const [termsError, setTermsError] = useState("");
+
+    const { error, signUpUser } = useAuth();
+    useDocumentTitle("Signup")
+
+    const emailMatchPattern =
+        /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
+
+    const signupHandler = () => {
+        if (!email.match(emailMatchPattern)) {
+            setEmailError(true);
+            return;
+        } else {
+            setEmailError(false);
+        }
+
+        if (password.length < 8) {
+            setPasswordError(true);
+            return;
+        } else {
+            setPasswordError(false);
+        }
+
+        if (!acceptTerms) {
+            setTermsError(true);
+            return;
+        } else {
+            setTermsError(false);
+        }
+
+        signUpUser(email, password);
+    };
+
+    return (
+        <main className="grand-main">
+            <div className="flex justify-content-center align-items-center h-100">
+                <div className="auth-container ">
+                    <h2 className="text-align-center font-black">Signup</h2>
+                    <div className="input-wrapper p-y-1 w-100">
+                        <label htmlFor="email" className="input-label fw-600">
+                            Email:
+                        </label>
+                        <input
+                            id="email"
+                            type="email"
+                            value={email}
+                            className={
+                                "input-field" +
+                                (emailError ? " input-color-error" : " ")
+                            }
```

Comment:
Suggestion: You can also convert this into a function "getInputClassName()" and pass the condition it will be easier in the future if you decide to change anything in the classname.

Reply:
will stick with your template literal suggestion. I think making a function for this would be unnecessary as the reader will have to search for what function does and even remember the reference so maybe it would be too much